src/model/search.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import _map from 'lodash/map';
+import _values from 'lodash/values';
+
+import { LEAST_SPANS, LONGEST_FIRST, MOST_RECENT, MOST_SPANS, SHORTEST_FIRST } from './order-by';
+import type { Trace } from '../types';
+import type { TraceSummaries, TraceSummary } from '../types/search';
+
+const isErrorTag = ({ key, value }) => key === 'error' && (value === true || value === 'true');
+
+/**
+ * Transforms a trace from the HTTP response to the data structure needed by the search page. Note: exported
+ * for unit tests.
+ *
+ * @param  {Trace} trace Trace data in the format sent over the wire.
+ * @return {TraceSummary} Summary of the trace data for use in the search results.
+ */
+export function getTraceSummary(trace: Trace): TraceSummary {
+  const { processes, spans, traceID } = trace;
+
+  let traceName = '';
+  let minTs = Number.MAX_SAFE_INTEGER;
+  let maxTs = Number.MIN_SAFE_INTEGER;
+  let numErrorSpans = 0;
+  // serviceName -> { name, numberOfSpans }
+  const serviceMap = {};
+
+  let i = 0;
+  for (; i < spans.length; i++) {
```

Comment:
Not familiar with this style? Is there an advantage than doing `for (let i = 0; i < spans.length; i++)`